sparse/src/KokkosSparse_BsrMatrix.hpp

```diff
@@ -451,12 +451,12 @@ class BsrMatrix {
     }
   }
 
-  /// \brief Constructor that copies raw arrays of host data in
-  ///   coordinate format.
+  /// \brief Construct BsrMatrix from host data in COO format.
   ///
-  /// On input, each entry of the sparse matrix is stored in val[k],
-  /// with row index rows[k] and column index cols[k].  We assume that
-  /// the entries are sorted in increasing order by row index.
+  /// The COO matrix must already have a block structure.
```

Comment:
@cwpearson, @lucbv: How do you normally populate in a COO matrix in production? Would a block format for crs2coo or ccs2coo help you?

Reply:
@e10harvey I stumbled across a couple things while working on this:

I'm not sure what this is doing, but I think it converts a CRS into BSR with the provided block size, filling in zeros as needed to make dense blocks:
https://github.com/kokkos/kokkos-kernels/blob/b483cfce31ab7ef98e484f922701f7d05fd186a5/sparse/src/KokkosSparse_Utils.hpp#L47

This actually does the block-formatted CRS to BSR through a different BSR constructor:
https://github.com/kokkos/kokkos-kernels/blob/b483cfce31ab7ef98e484f922701f7d05fd186a5/sparse/src/KokkosSparse_Utils.hpp#L208

In any case, both are only used in perf and unit tests.

In practice, I suspect that the apps generate block-structured matrices in memory on the fly, so they never hit the disk except for checkpoints. Our Kernels utilities for handling these transformations are all in host memory and probably uniformly slow, so if apps really need it they've probably rolled their own by now. Maybe @lucbv knows how useful it might be for us to optimize these sorts of things